converter.go

```diff
@@ -179,19 +178,34 @@ func resizeNRGBA(in *image.NRGBA, out *image.NRGBA, scale float64, coeffs []int1
 					default:
 						xi = 0
 					}
-					rgba[0] += int32(coeff) * int32(row[xi+0])
-					rgba[1] += int32(coeff) * int32(row[xi+1])
-					rgba[2] += int32(coeff) * int32(row[xi+2])
-					rgba[3] += int32(coeff) * int32(row[xi+3])
+
+					// Forward alpha-premultiplication
+					a := int32(row[xi+3])
+					rgba[0] += int32(coeff) * int32(row[xi+0]) * a / 0xff
+					rgba[1] += int32(coeff) * int32(row[xi+1]) * a / 0xff
+					rgba[2] += int32(coeff) * int32(row[xi+2]) * a / 0xff
+					rgba[3] += int32(coeff) * a
```

Comment:
This is doing both: Assigning color values and performing the alpha-premultiplication. To keep things clear, these should be separated. Also the steps in the premultiplication should be more explicit to make them easier to understand, like
```
rgba[0] *= a
rgba[0] /= 0xff
```
similar to how it's done in https://golang.org/src/image/color/color.go